bin/s2nc.c

```diff
@@ -48,18 +48,37 @@ void usage()
     fprintf(stderr, "  --name [server name]\n");
     fprintf(stderr, "    Sets the SNI server name header for this client.  If not specified, the host value is used.\n");
     fprintf(stderr, "\n");
-    fprintf(stderr, "  --s,--status\n");
+    fprintf(stderr, "  -s,--status\n");
     fprintf(stderr, "    Request the OCSP status of the remote server certificate\n");
     fprintf(stderr, "  --mfl\n");
     fprintf(stderr, "    Request maximum fragment length from: 512, 1024, 2048, 4096\n");
+    fprintf(stderr, "  -f,--ca-file [file path]\n");
+    fprintf(stderr, "    Location of trust store CA file (PEM format).\n");
+    fprintf(stderr, "  -d,--ca-dir [directory path]\n");
+    fprintf(stderr, "    Directory containing hashed trusted certs..\n");
     fprintf(stderr, "\n");
     exit(1);
 }
 
+struct verify_data {
+    const char *trusted_host;
+};
+
+static uint8_t verify_host(const char *host_name, size_t host_name_len, void *data) {
+    struct verify_data *verify_data = (struct verify_data *)data;
+
+    char *offset = strstr(host_name, "*.");
+    if(offset) {
+        return (uint8_t)(strstr(verify_data->trusted_host, offset + 2) != NULL);
```

Comment:
it is case-sensitive. No it doesn't. Neither does any other TLS implementation I know of. 

I didn't put much thought into this one since it is in the console driver, and it's more useful as a sample for how to use the x509 validation.

Do you disagree?

Reply:
nginx does case insensitive match on hostnames with certs.  i think case insensitive makes more sense, as DNS is case insensitive and it makes sense to be consistent here.